packages/node-integration-tests/suites/tracing/tracePropagationTargets/test.ts

```diff
@@ -0,0 +1,37 @@
+import nock from 'nock';
+
+import { runScenario, runServer } from '../../../utils';
+
+test('HttpIntegration should instrument correct requests when tracePropagationTargets option is provided', async () => {
+  const match1 = nock('http://match-this-url.com')
+    .get('/api/v0')
+    .matchHeader('baggage', val => typeof val === 'string')
+    .matchHeader('sentry-trace', val => typeof val === 'string')
+    .reply(200);
+
+  const match2 = nock('http://match-this-url.com')
+    .get('/api/v1')
+    .matchHeader('baggage', val => typeof val === 'string')
+    .matchHeader('sentry-trace', val => typeof val === 'string')
+    .reply(200);
+
+  const match3 = nock('http://dont-match-this-url.com')
+    .get('/api/v2')
+    .matchHeader('baggage', val => val === undefined)
+    .matchHeader('sentry-trace', val => val === undefined)
+    .reply(200);
+
+  const match4 = nock('http://dont-match-this-url.com')
+    .get('/api/v3')
+    .matchHeader('baggage', val => val === undefined)
+    .matchHeader('sentry-trace', val => val === undefined)
+    .reply(200);
+
+  const serverUrl = await runServer(__dirname);
+  await runScenario(serverUrl);
+
+  expect(match1.isDone()).toBe(true);
```

Comment:
Not a nock expert; What does `isDone()` tell us? 

Or more generally, the way I understand these integration tests, we first run this test file and we set up the nock endpoints, right? Then the scenario is executed and we check that the headers are in/excluded in the respective nock endpoints. Is this what's happening?

Reply:
So every `nock()` call is just setting up an "interception" for one request that matches the configuration of the nock call. In this case, the configurations are a specific URL with a specific path, and the (non-)existance of a set of headers.

Asserting on isDone() is basically just checking whether the interceptors "fired". Boiled down, we're asserting that a specific set of http calls have been made.

Reply:
I see, makes sense, thanks!